src/lib.rs

```diff
@@ -11,7 +11,15 @@ use router::{Match, Router};
 
 #[derive(Default)]
 pub struct RouteBuilder {
-    routers: HashMap<Method, Router<Box<dyn Handler>>>,
+    routers: HashMap<Method, Router<WrappedHandler>>,
+}
+
+#[derive(Clone)]
+struct RoutePattern(&'static str);
+
+pub struct WrappedHandler {
+    pattern: RoutePattern,
+    handler: Box<dyn Handler>,
 }
```

Comment:
To build crates.io with this PR, I had to add the following impl to this struct:

```rust
impl conduit::Handler for WrappedHandler {
    fn call(&self, request: &mut dyn RequestExt) -> HandlerResult {
        self.handler.call(request)
    }
}
```

to avoid:

```
error[E0599]: no method named `call` found for reference `&WrappedHandler` in the current scope
   --> src/router.rs:179:27
    |
179 |                 m.handler.call(req)
    |                           ^^^^ method not found in `&WrappedHandler`
```